src/v/security/jwt.h

```diff
@@ -0,0 +1,623 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+#pragma once
+
+#include "json/document.h"
+#include "oncore.h"
+#include "outcome.h"
+#include "utils/string_switch.h"
+#include "utils/utf8.h"
+
+#include <seastar/core/sstring.hh>
+#include <seastar/util/variant_utils.hh>
+
+#include <absl/algorithm/container.h>
+#include <absl/container/flat_hash_map.h>
+#include <boost/algorithm/string/split.hpp>
+#include <cryptopp/base64.h>
+#include <cryptopp/cryptlib.h>
+#include <cryptopp/integer.h>
+#include <cryptopp/osrng.h>
+#include <cryptopp/rsa.h>
+
+#include <optional>
+#include <string_view>
+#include <system_error>
+
+namespace security::oidc {
+
+enum errc {
+    success = 0,
+    metadata_invalid,
+    jwks_invalid,
+    jwk_invalid,
+    jws_invalid,
+    jwt_invalid,
+    kid_not_found,
+};
+
+struct errc_category final : public std::error_category {
+    const char* name() const noexcept final { return "security::oidc::errc"; }
+
+    std::string message(int c) const final {
+        switch (static_cast<errc>(c)) {
+        case errc::success:
+            return "Success";
+        case errc::metadata_invalid:
+            return "Invalid metadata";
+        case errc::jwks_invalid:
+            return "Invalid jwks";
```

Comment:
what is jwks - jason web key service? or smth.

Reply:
> what is jwks - jason web key service? or smth. 

JSON Web Key Set